auth/credentials/idtoken/idtoken.go

```diff
@@ -78,9 +78,25 @@ type Options struct {
 
 	// CredentialsFile sources a JSON credential file from the provided
 	// filepath. If provided, do not provide CredentialsJSON. Optional.
+	//
+	// Important: If you accept a credential configuration (credential
+	// JSON/File/Stream) from an external source for authentication to Google
+	// Cloud Platform, you must validate it before providing it to any Google
+	// API or library. Providing an unvalidated credential configuration to
+	// Google APIs can compromise the security of your systems and data. For
+	// more information, refer to [Validate credential configurations from
+	// external sources](https://cloud.google.com/docs/authentication/external/externally-sourced-credentials).
 	CredentialsFile string
 	// CredentialsJSON sources a JSON credential file from the provided bytes.
 	// If provided, do not provide CredentialsJSON. Optional.
+	//
+	// Important: If you accept a credential configuration (credential
+	// JSON/File/Stream) from an external source for authentication to Google
+	// Cloud Platform, you must validate it before providing it to any Google
+	// API or library. Providing an unvalidated credential configuration to
+	// Google APIs can compromise the security of your systems and data. For
+	// more information, refer to [Validate credential configurations from
+	// external sources](https://cloud.google.com/docs/authentication/external/externally-sourced-credentials).
```

Comment:
Does impersonated cred also need a similar update then?

Reply:
I'm not sure. In this PR I did not update configuration fields that accept the general type `auth.Credentials`. PTAL at the `impersonate` package surface and let me know if you see any configuration fields (or functions) that need the warning, thank you. https://pkg.go.dev/cloud.google.com/go/auth@v0.14.0/credentials/impersonate

Reply:
I don't believe the `impersonate` package is used with external accounts. I think impersonation use cases with external accounts are supported by the `externalaccount` package, which is updated in this PR in `auth/credentials/externalaccount/externalaccount.go`.